plugin/proc_info/proc_info.cc

```diff
@@ -0,0 +1,77 @@
+#include <my_global.h>
+#include <table.h>
+#include <sql_show.h>
+#include <sql_class.h>
+#include <mysql/plugin.h>
+
+
+static ST_FIELD_INFO meminfo_fields[]=
+{
+  { "NAME", 100, MYSQL_TYPE_STRING, 0, 0, 0, SKIP_OPEN_TABLE },
+  { "VALUE", 21, MYSQL_TYPE_LONG, 0, MY_I_S_UNSIGNED, 0, SKIP_OPEN_TABLE },
+  { 0, 0, MYSQL_TYPE_STRING, 0, 0, 0, 0 }
+};
+
+
+static int meminfo_fill(MYSQL_THD thd, TABLE_LIST *tables, COND *cond)
+{
+  TABLE *table= tables->table;
+  char name[1024];
+  unsigned long value;
+  FILE *fp;
```

Comment:
MYSQL_TYPE_LONG is int32. Probably not sufficient to describe virtual memory, even in kilobytes.